_drafts/zio-fibers.md

```diff
@@ -0,0 +1,290 @@
+
+```sbt
+libraryDependencies += "dev.zio" %% "zio" % "1.0.9"
+```
+
+Nothing else will be required.
+
+## 2. The Effect Pattern
+
+Before talking about ZIO fibers, we need to have the notion of the _Effect Pattern_. The main objective of any effect library is to deal with statements that can lead to side effects. The _substitution model_ is the fundamental building block of functional programming. Unfortunately, **we cannot model side effects using the substitution model because we cannot substitute functions with  their results**, varying from information other than simple functions' inputs. Such functions are often called _impure_:
+
+```scala
+// The type of println is String => Unit. The program prints the given String to the console
+val result = println("The meaning of life is 42")
+// Using the substitution model, we try to substitute the result of the println execution to the variable
+val result: Unit = ()
+//...however, after the substitution, the meaning of the program completely changed
+```
+
+So, it comes the Effect Pattern. The pattern aims to model side effects with the concept of effect. **An effect is a blueprint of statements that can produce a side effect, not the result itself**. So, when we instantiate an effect with some statements, we don't execute anything: We are just describing what the code inside the effect will perform once executed:
+
+```scala
+// This code doesn't print anything to the console. It's just a blueprint
+val result = ZIO.succeed(println("The meaning of life is 42"))
+```
+
+
+The above code doesn't print anything to the console; it just describes what we want to achieve with the code, not how we execute it.
+
+Moreover, the Effect Pattern adds two more conditions to the whole story:
+
+1. The type of the effect must describe the type returned by the effect once executed and what kind of side effect it represents.
+2. The use of the effect must separate the description of the side effect (the blueprint) from its execution.
+
+The first condition allows us to trace the impure code in our program. One of the main problems with side effects is that they are hidden in the code. Instead, the second condition allows us to use the substitution model until we need to run the effect. We call such a point "the end of the world", and we merely identify it with the `main` method.
+
+## 3. ZIO Effect
+
+The ZIO library is one of the implementations of the effect pattern available in Scala. Its primary effect type is `ZIO[R, E, A]`. We call the `R` type the _environment type_, representing the dependencies the effect needs to execute. Once executed, the effect can produce a result of type `A` or fail, producing a value of the type `E`. For these reasons, it's common to think about the type`ZIO[R, E, A]` as an _effectful_ version of the function type `R => Either[E, A]`.
```

Comment:
Describe what "effectful" means:

"... i.e. the evaluation of a ZIO might produce a side effect."